fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java

```diff
@@ -120,14 +121,22 @@ public HivePartition load(PartitionCacheKey key) throws Exception {
                     }
                 }, executor));
 
-        fileCache = CacheBuilder.newBuilder().maximumSize(Config.max_external_file_cache_num)
-                .expireAfterAccess(Config.external_cache_expire_time_minutes_after_access, TimeUnit.MINUTES)
-                .build(CacheLoader.asyncReloading(new CacheLoader<FileCacheKey, ImmutableList<InputSplit>>() {
-                    @Override
-                    public ImmutableList<InputSplit> load(FileCacheKey key) throws Exception {
-                        return loadFiles(key);
-                    }
-                }, executor));
+        // if the file.meta.cache.ttl-second is equal or greater than 0, the cache expired will be set to that value
+        int fileMetaCacheTtlSecond = NumberUtils.toInt(
+            (catalog.getProperties().getOrDefault(HMSExternalCatalog.FILE_META_CACHE_TTL_SECOND, "")), HMSExternalCatalog.FILE_META_CACHE_NO_TTL);
+
+        CacheBuilder<Object, Object> fileCacheBuilder = CacheBuilder.newBuilder().maximumSize(Config.max_external_file_cache_num);
```

Comment:
If `file.meta.cache.ttl-second` = '0', maybe we should not initialize the loading cache, just set `fileCache` = null and let the current thread to load the file splits. Because loading file splits of a hive table with many partitions is a time-consuming task, it can easily drain the cache loader thread pool(pool size = 10 now),  it will block all querys with external tables if any cache (partitionValuesCache/partitionCache/fileCache/schemaCache) is expired.

Reply:
When duration is zero, this method hands off to maximumSize(0), ignoring any otherwise-specified maximum size or weight. This can be useful in testing, or to disable caching temporarily without a code change.

Reply:
1.

Reply:
Yeah, well, thanks for your advice. I agree with you. To handle the code more gracefully, I recommend that the guava CacheLoader be set to synchronous.

Reply:
I think this is a good idea~

Reply:
This pr is lgtm, but my perspective is limited, maybe you can request `morinngman` to review it again.